examples/snake.rs

```diff
@@ -0,0 +1,324 @@
+extern crate coffee;
+
+use coffee::graphics::{
+    Color, Font, Frame, Mesh, Point, Rectangle, Shape, Text, Window, WindowSettings,
+};
+use coffee::input::keyboard::KeyCode;
+use coffee::input::{self, keyboard, Input};
+
+use coffee::load::Task;
+use coffee::{Game, Timer};
+
+use rand::seq::IteratorRandom;
+
+fn main() {
+    SnakeGame::run(WindowSettings {
+        title: String::from("Snake"),
+        size: (900, 600),
+        resizable: false,
+        maximized: false,
+        fullscreen: false,
+    })
+    .expect("An error occured while starting the game");
+}
+
+fn new_random_pos() -> (f32, f32) {
+    let mut rng = rand::thread_rng();
+    let x = (0..900).step_by(30).choose(&mut rng).unwrap() as f32;
+    let y = (0..600).step_by(30).choose(&mut rng).unwrap() as f32;
+    return (x, y);
+}
+#[derive(Debug, Clone, Copy, PartialEq)]
+struct Position(f32, f32);
+
+struct Snake {
+    square_pos: Vec<Position>,
+    direction: Option<KeyCode>,
+}
+
+impl Snake {
+    fn new() -> Snake {
+        Snake {
+            square_pos: vec![],
+            direction: None,
+        }
+    }
+    fn create_snake(&mut self) {
+        for i in 1..6 {
+            let x = (30.0 * i as f32) + 30.0;
+            let y = 90.0;
+            self.square_pos.push(Position(x, y));
+        }
+    }
```

Comment:
We could move this logic to `new`. This way we avoid the scenario of "empty snakes".